From 3deb588ec3b3567a4680d6bfd77cc5042ad336e7 Mon Sep 17 00:00:00 2001
From: Jayachandran C <jayachandranc@google.com>
Date: Tue, 2 Sep 2025 05:59:26 +0000
Subject: [PATCH] Protect shell overriding the carrier config

Bug: 441823943
Flag: EXEMPT Bugfix
Test: Manually verified the scenario and all telephony and telecom CTS are verified
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:1ac1e79d18ba13996466fb79fa1c665b9ee9b5b8)
Merged-In: I2e5d5b59dd95aa60a07896bf335ebe14c97c411a
Change-Id: I2e5d5b59dd95aa60a07896bf335ebe14c97c411a
---
 src/com/android/phone/CarrierConfigLoader.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/com/android/phone/CarrierConfigLoader.java b/src/com/android/phone/CarrierConfigLoader.java
index 47fd96e8e..358bfd9e1 100644
--- a/src/com/android/phone/CarrierConfigLoader.java
+++ b/src/com/android/phone/CarrierConfigLoader.java
@@ -1437,6 +1437,13 @@ public class CarrierConfigLoader extends ICarrierConfigLoader.Stub {
     public void overrideConfig(int subscriptionId, @Nullable PersistableBundle overrides,
             boolean persistent) {
         overrideConfig_enforcePermission();
+
+        // Do not allow shell UID to override the carrier config. This will not impact
+        // the CTS and telephony shell commands as they use different uids
+        if (TelephonyPermissions.isShell(getCallingUid())) {
+            throw new SecurityException("overrideConfig cannot be invoked by shell");
+        }
+
         int phoneId = SubscriptionManager.getPhoneId(subscriptionId);
         if (!SubscriptionManager.isValidPhoneId(phoneId)) {
             logd("Ignore invalid phoneId: " + phoneId + " for subId: " + subscriptionId);
-- 
2.51.2.1006.ga50a493c49-goog

